[Lim Junxue] iP

LimJunxue opened the pull request.

Commit:
Add Gradle support

Commit:
build.gradle: Update version to 8.29

LimJunxue committed:
Level-1 Greet, Echo, Exit

LimJunxue committed:
Level-2 Add, List

LimJunxue committed:
Level-3 Mark as done

LimJunxue committed:
Level-4 ToDo, Event, Deadline

LimJunxue committed:
A-TextUiTesting

LimJunxue committed:
Level-5 Handle Errors

LimJunxue committed:
Level-5 rename Task

LimJunxue committed:
Level-5 rename tasks in other files

LimJunxue committed:
Level-6 Delete

LimJunxue committed:
A-Enums

didn't actually implement enums

LimJunxue committed:
Create storage class

LimJunxue committed:
Add functionality for updating storage

Level-7

LimJunxue committed:
Store deadline dates as LocalDate

LimJunxue committed:
Merge branch 'branch-Level-7' into master

LimJunxue committed:
Merge branch 'branch-Level-8' into master

LimJunxue committed:
Refactor Duke for more OOP

Create Parser, Ui and TaskList classes to abstract the activities of Duke

LimJunxue committed:
Package classes into DukeObjects

LimJunxue committed:
Create JUnit test classes for Parser and Storage

LimJunxue committed:
Repackage Duke and other objects in duke

LimJunxue committed:
Comment JavaDoc comments

LimJunxue committed:
Configure IntelliJ to coding standards and reformat

LimJunxue committed:
Create find functionality Level-9

LimJunxue committed:
Merge branch 'branch-A-JavaDoc' into master

LimJunxue committed:
Merge branch 'branch-A-CodingStandard' into master

LimJunxue committed:
Merge branch 'branch-Level-9' into master

LimJunxue committed:
Correct README

LimJunxue committed:
Merge branch 'master' into master

LimJunxue committed:
Merge branch 'add-gradle-support'

LimJunxue committed:
Test gradle

LimJunxue committed:
Add checkstyle

Sidney011100 commented on src/main/java/duke/Deadline.java:
```diff
@@ -0,0 +1,63 @@
+     */
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
```
could there be a more meaningful variable name as compared to "by"?

Sidney011100 commented on src/main/java/duke/DukeActions.java:
```diff
@@ -0,0 +1,24 @@
+    /**
+     * The String form of each action.
+     */
+    public final String str;
```
maybe str could be changed to action to prevent the use of shortforms, and for increased readability

Sidney011100 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,150 @@
+                Ui.printErr("Please enter a number from 1 to " + tasks.size() + " after delete!");
+            }
+        }
+    }
```
Overall very easy-to-read codes, however would explicitly writing out the variable names like currTask and printErr improve the readability?

Sidney011100 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,144 @@
+        try {
+            Scanner reader = new Scanner(file);
+            while (reader.hasNextLine()) {
+                boolean isDone = reader.nextInt() == 1;
```
a bit of an "arrowhead" code here, maybe can create another method for code in the while loop

Sidney011100 commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,114 @@
+     * @throws DukeException if there is no description of the task.
+     * @throws DukeException if there is no deadline specified by "/by".
+     */
+    private static Task createDeadline(String command) throws DukeException {
```
would it be better if each subclass had a create() method?

Sidney011100 commented on src/test/java/StorageTest.java:
```diff
@@ -0,0 +1,48 @@
+    void create_new_file() {
+        Storage.createStorage(path);
+        assertTrue(new File(path).exists());
+    }
```
maybe the convention of naming test methods could be checked up

DineshMagesvaran commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,24 @@
+ * Represents a task with a duration specified in the "at" argument.
+ */
+public class Event extends Task {
+    protected String at;
```
Is there a more descriptive variable name that could be better for 'at'?

DineshMagesvaran commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,150 @@
+    }
+
+    /**
+     * Passing the command to the relevant methods.
```
It is recommended for the method header to start with 'passes' instead of 'passed'.

DineshMagesvaran commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,150 @@
+        try {
+            String query = command.split(" ", 2)[1].trim();
+            StringBuffer reply = new StringBuffer();
+            boolean found = false;
```
'isFound' may be a better alternative to 'found' as it is a boolean variable.

DineshMagesvaran commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,98 @@
+        if (command.length() > 0) {
+            throw new IllegalArgumentException("That is not a valid command!\n" +
+                    "These are the possible commands:\n" +
+                    "\"list\" \"done\" \"delete\" \"todo\" \"deadline\" \"event\" \"bye\"");
```
Well written error message as it is very descriptive and tells the user what to do.

DineshMagesvaran reviewed (commented):
Overall, the code is very well-written and easy to read. The code also follows the java coding standards quite well.

LimJunxue commented:
Thank you @Sidney011100 and @DineshMagesvaran for your comments! I will implement them soon.

LimJunxue committed:
Apply peer review comments

LimJunxue committed:
Implement JavaFX as GUI

Apply JavaFX to make a GUI, showing a conversation.
Make DialogBox and MainWindow fxml controller files.
Change all CLI output into GUI text.

LimJunxue committed:
Format minor stuff

LimJunxue committed:
Merge branch 'branch-Level-10' into master

LimJunxue committed:
Write assertions for Parser and MainWindow

Assert that duke was created and set before showing welcome. Assert that command handlers in Parser were called correctly.

LimJunxue committed:
Improve code quality and repackage task related classes

LimJunxue committed:
Merge pull request #1 from LimJunxue/branch-A-Assertions

Write assertions for Parser and MainWindow

LimJunxue committed:
Merge branch 'master' into branch-A-CodeQuality

LimJunxue committed:
Merge pull request #2 from LimJunxue/branch-A-CodeQuality

Improve code quality and repackage task related classes

LimJunxue committed:
Create gradle.yml

LimJunxue committed:
Make reminders function

Make reminders function that shows all deadline tasks

LimJunxue committed:
Merge branch 'branch-BCD-Extension' into master

LimJunxue committed:
Change folder of yml file

LimJunxue committed:
Fix gradle checks for CI

LimJunxue committed:
Reduce dialog padding on top and bottom

LimJunxue committed:
Credit se-edu

LimJunxue committed:
Credit se-edu

LimJunxue committed:
Add a user guide and Ui picture

LimJunxue committed:
Change app main class name in build.gradle

LimJunxue committed:
Improve GUI

Add icon and rounder text box.

LimJunxue committed:
Update README and source
